### PR TITLE
docs: new documentation added at deploying-with-pm2-and-nginx.md, con…

### PR DESCRIPTION
**Pull request by vvdwivedi**
Added documentation for Deploying using pm2 behind a reverse proxy. 

Implements #1850

## Checklist

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

**Comment by vvdwivedi** [comment]
The AppVeyor build failure doesn't seem to be coming from this commit. It was just a doc added.

**Comment by bajtos** [comment]
@vvdwivedi thank you for the pull request, it's great to see more guides coming from our community ❤️ 

I'll review the content a bit later, but: please add the new page to the left sidebar, the section "How-tos". Just copy, paste and modify an entry for a similar page, e.g. "Deploying to IBM Cloud".

https://github.com/strongloop/loopback-next/blob/8156f9d32aac3a0461b617a01cde9c319a1cceca/docs/site/sidebars/lb4_sidebar.yml#L194-L212

Learn more here: https://loopback.io/doc/en/contrib/sidebar_navigation.html#sidebar-syntax

File: docs/site/Deploying-with-pm2-and-nginx.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+```sh
+$ lb4 app
+```
+
+Refer to
+[getting started](https://loopback.io/doc/en/lb4/Getting-started.html#create-a-new-project)
+section of documentation for detailed instructions.
+
+You also need to have pm2 installed globally. If not you don't already have pm2
```
Comment (bajtos) [inline docs/site/Deploying-with-pm2-and-nginx.md]: It a global install something that `pm2` recommends? In my experience, it's usually better to use a local install and invoke `pm2` via npm script.

```json
{
  "scripts": {
    "start": " pm2 start index.js"
  },
  "dependencies": {
    "pm2": "^1.2.3"
  }
}
```

That way each app retains control over which version of `pm2` it is working with, and developers working on multiple apps don't run into a possible conflict in the required version of the global `pm2` install.

**Comment by bajtos** [comment]
@rmg @sam-github IIRC, you used to have a lot of knowledge about deploying Node.js apps and running them under a supervisor behind a reverse proxy. Could you please help us to review these proposed instructions for running a LoopBack4 application using pm2+nginx?

File: docs/site/Deploying-with-pm2-and-nginx.md (hunk 1 above)
Comment (bajtos) [reply docs/site/Deploying-with-pm2-and-nginx.md]: Sounds good, thanks 👍

**Comment by bajtos** [comment]
> Merge branch 'master' into pm2-deployment

We like to keep our git history linear and don't do merge commits. Please run `git rebase master && git push --force-with-lease` instead. (GitHub's "Update branch" button creates a merge commit, therefore it cannot be used either.)

File: docs/site/Deploying-with-pm2-and-nginx.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+### pm2 setup
+
+> > You might find yourself in a situation in which you do not have access to
+> > the CLI to start your Node.js applications. In such a situation, pm2 must be
+> > added as a dependency and must be called with the start script.
+
+1. Generate an ecosystem.config.js template with:
+
+```sh
```
Comment (bajtos) [inline docs/site/Deploying-with-pm2-and-nginx.md]: When you don't indent a paragraph, then you are telling Markdown to put it at to top level.

I think your intention was to render this code block as part of the first item, wasn't it?

To achieve that, you need to indent the entire block by four spaces.

The same comment applies to most of the content below.

File: docs/site/Deploying-with-pm2-and-nginx.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+$ pm2 init
+```
+
+2. Modify the generated file to match loopback requirements:
+
+```
```
Comment (bajtos) [inline docs/site/Deploying-with-pm2-and-nginx.md]: Please set the file type to `js` to enable code highlighting. See https://loopback.io/doc/en/contrib/code_samples.html#code-samples

Also, what is the name of the generated file? Can you add a code caption please? See https://loopback.io/doc/en/contrib/code_samples.html#code-caption

File: docs/site/Deploying-with-pm2-and-nginx.md (hunk 2 above)
Comment (vvdwivedi) [reply docs/site/Deploying-with-pm2-and-nginx.md]: right, I noticed something is off, but wasn't aware of this. Fixed at all parts of docs.

File: docs/site/Deploying-with-pm2-and-nginx.md (hunk 3 above)
Comment (vvdwivedi) [reply docs/site/Deploying-with-pm2-and-nginx.md]: Done

Comment (vvdwivedi) [reply docs/site/Deploying-with-pm2-and-nginx.md]: Done

File: docs/site/Deploying-with-pm2-and-nginx.md (hunk 2 above)
Comment (vvdwivedi) [reply docs/site/Deploying-with-pm2-and-nginx.md]: Done. I saw something off, wasn't aware of this.

File: docs/site/Deploying-with-pm2-and-nginx.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+$ lb4 app
+```
+
+Refer to
+[getting started](https://loopback.io/doc/en/lb4/Getting-started.html#create-a-new-project)
+section of documentation for detailed instructions.
+
+### pm2 setup
+
+> > You might find yourself in a situation in which you do not have access to
```
Comment (bajtos) [inline docs/site/Deploying-with-pm2-and-nginx.md]:
```suggestion
> You might find yourself in a situation in which you do not have access to
```

File: docs/site/Deploying-with-pm2-and-nginx.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+> > the CLI to start your Node.js applications. In such a situation, pm2 must be
```
Comment (bajtos) [inline docs/site/Deploying-with-pm2-and-nginx.md]:
```suggestion
> the CLI to start your Node.js applications. In such a situation, pm2 must be
```

File: docs/site/Deploying-with-pm2-and-nginx.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+> > added as a dependency and must be called with the start script.
```
Comment (bajtos) [inline docs/site/Deploying-with-pm2-and-nginx.md]:
```suggestion
> added as a dependency and must be called with the start script.
```

File: docs/site/Deploying-with-pm2-and-nginx.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+1. Generate an ecosystem.config.js template with:
+
+   ```sh
+   $ pm2 init
+   ```
+
+2. Modify the generated file to match loopback requirements:
+
+   {% include ecosystem.config.js content="/ecosystem.config.js" %}
```
Comment (bajtos) [inline docs/site/Deploying-with-pm2-and-nginx.md]:
```suggestion
   {% include code-caption.html content="/ecosystem.config.js" %}
```

Comment (bajtos) [reply docs/site/Deploying-with-pm2-and-nginx.md]: I think this line is causing the build failure.

File: docs/site/Deploying-with-pm2-and-nginx.md (hunk 4 above)
Comment (vvdwivedi) [reply docs/site/Deploying-with-pm2-and-nginx.md]: Done

File: docs/site/Deploying-with-pm2-and-nginx.md (hunk 5 above)
Comment (vvdwivedi) [reply docs/site/Deploying-with-pm2-and-nginx.md]: Done

File: docs/site/Deploying-with-pm2-and-nginx.md (hunk 6 above)
Comment (vvdwivedi) [reply docs/site/Deploying-with-pm2-and-nginx.md]: Done

File: docs/site/Deploying-with-pm2-and-nginx.md (hunk 7 above)
Comment (vvdwivedi) [reply docs/site/Deploying-with-pm2-and-nginx.md]: Yes, I was not very clear from the documentation about `code-caption.html`. Ran the checks and it works fine.

Comment (bajtos) [reply docs/site/Deploying-with-pm2-and-nginx.md]: Would you mind improving our docs to make it more clear for future readers?

**Comment by bajtos** [comment]
Uh oh, it looks like the rebase did not go well, your changes include `hasOne` implementation too. Let me try to fix the git history for you.

**Comment by bajtos** [comment]
Running this command fixed the history: `git rebase master`. Note that I force-pushed into your feature branch, don't do `git pull` now please.

**Comment by bajtos** [comment]
Landed, thank you @vvdwivedi for the contribution!

File: docs/site/Deploying-with-pm2-and-nginx.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+         },
+         env_production: {
+           NODE_ENV: 'production',
+         },
+       },
+     ],
+   };
+   ```
+
+3. Add pm2 as a dependency to your projet using following command
```
Comment (sam-github) [inline docs/site/Deploying-with-pm2-and-nginx.md]: "project"?

File: docs/site/Deploying-with-pm2-and-nginx.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+## Deployment
+
+1. Register and start your application with pm2 using following command at the
+   app root directory:
+
+   ```sh
+   $ npm start
+   ```
+
+   This creates a dist folder which containes the transpiled code. At you app
```
Comment (sam-github) [inline docs/site/Deploying-with-pm2-and-nginx.md]: "your app's"

**Comment by b-admike** [comment]
@vvdwivedi Thank you for your contribution! Would you like to address @sam-github's comments in a new PR?

**Pull request merged**